### ISSUE TITLE
sql: sys table for checkpoints

### ISSUE THREAD

**Issue by vivekmenezes** [cockroachdb/cockroach#6665]
We have long running workloads they need to checkpoint every once in a while. The schema change mechanism operating on large tables is one such use case. Long running distributed queries, backups for example, will certainly need to checkpoint. These workloads are supervised from a single node, and when that supervisor dies, the entire workload needs to start from afresh. What we need is a way for the supervisor to store its state, restart from a saved state, and a GC of the old unused state.

```
    checkpointTableSchema = `
CREATE TABLE system.checkpoint (
    key         STRING PRIMARY KEY,
    checkpoint       BYTES NOT NULL,
    lastUpdated TIMESTAMP NOT NULL
);
```

**Comment by vivekmenezes** [comment]
To allow existing clusters to operate well with this change, we need to add a mechanism for the server to create the table outside of the bootstrap process.

There are two ways that can be done:
1. Have the server create a pgwire connection with itself and attempt to create the table at startup.
2. The server acts like the bootstrap process and directly writes the table descriptor to the kv store using CPut.

I would normally think that 1 is the better option but that requires creating a pgwire connection with certs to the server for this particular use case and I'm not sure whether it's wise from a security standpoint. @mberhault care to comment.
1 also has the added advantage that it can be used to rollout a new schema change to an existing system table.

**Comment by bdarnell** [comment]
Internally we should use `sql.InternalExecutor` so we can use the node certificates at the KV level instead of going through the external sql interface.

**Comment by mberhault** [comment]
isn't there also option:
3. Plain sql statement (eg: `CREATE TABLE IF NOT EXISTS system.checkpoint (...)`) called internally (hand it off directly to SQL, but not using a pg client)

**Comment by tamird** [comment]
#4869

**Comment by vivekmenezes** [comment]
okay I'll go with the internal sql implementation. Thanks

**Comment by vivekmenezes** [comment]
@mberhault  I was planning on using user: NodeUser for the internal schema changes to the system tables. Is that okay? I have a hacked up version working that uses user: RootUser

**Comment by vivekmenezes** [comment]
fixed by #13995
